core/src/main/java/com/xatkit/core/recognition/RecognitionMonitorInflux.java

```diff
@@ -0,0 +1,600 @@
+     */
+    private char[] token;
+
+    /**
+    * The {@link Configuration} key to specify a custom bucket instance to store the analytics.
+    */
+    static final String INFLUX_BUCKET_KEY = "xatkit.influx.bucket";
+
+    /**
+     * The BUCKET value specified in the {@link Configuration}, necessary for the petitions to the database.
+     */
+    private String bucket;
+
+    /**
+     * The default value for BUCKET
+     */
+    private static final String DEFAULT_BUCKET = "xatbot";
+
+    /**
+    * The {@link Configuration} key to specify a custom organization workspace for influx.
+    */
+    static final String INFLUX_ORG_KEY = "xatkit.influx.organization";
+
+    /**
+     * The ORGANIZATION value specified in the {@link Configuration}, necessary for the petitions to the database.
+     */
+    private String organization;
+
+    /**
+     * The default value for ORGANIZATION
+     */
+    private static final String DEFAULT_ORGANIZATION = "Xatkit";
+
+    /**
+     * The databases url key.
+     * <p>
+     * This property is optional and will use "http://localhost:7777" as default
+     */
+    private static final String INFLUX_URL_KEY = "xatkit.influx.url";
+
+    /**
+     * The default value for the URL of influx instance
+     */
+    private static final String DEFAULT_URL = "http://localhost:7777";
+
+    /**
+     * The default ID of the bot
+     */
+    private static final String DEFAULT_BOT_ID = "xatkitBot";
+
+    /**
+     * The actual ID of the bot
+     */
+    private String bot_Id;
+
+    /**
+     * The database persistent client to make the petitions :)
+     */
+     private InfluxDBClient db;
+
+    /**
+    *   Influxdb stores data, by default in "/var/lib/influxdb/wal" or "/var/lib/influxdb/data"
+    *   based on it's configuration: wal files are "temporal" until they reach 25MB (default config)
+    *   or when the data has been there for 10 minutes. This can be changed in the [data] section from influxdb.conf
+    *   To be able to query/input data with influx we need the following information:
+    *   TOKEN: this validates our bot into the database, so it will be authenticated.
+    *   BUCKET: named location where data is stored. It has a retention policy, a duration of time that each data point persists, etc.
+    *   A bucket belongs to 1 organization.
+    *   ORGANIZATION:        it's the workspace/group of users. It can own multiple buckets.
+    *   @param xatkitServer  the {@link XatkitServer} instance used to register the REST endpoints
+    *   @param configuration the Xatkit {@link Configuration}
+    */
+    public RecognitionMonitorInflux(XatkitServer xatkitServer, Configuration configuration){
+        Log.info("Starting new intent recognition monitoring with Influxdb");
+        // Required for storing/querying data from influx:
+        token = configuration.getString(INFLUX_TOKEN_KEY).toCharArray();
```

Comment:
This crashes with a NPE if the configuration doesn't contain the token. Would be better to check the value and throw a comprehensible exception if it is null.
I'll take care of that.